init_tinygo.go

```diff
@@ -5,13 +5,28 @@
 
 package main
 
-import _ "github.com/wasilibs/nottinygc"
+import (
+	"unsafe"
 
-// Compiled by nottinygc for delayed free but Envoy doesn't stub it yet,
-// luckily nottinygc doesn't actually call the function, so it's fine to
-// stub it out.
+	_ "github.com/wasilibs/nottinygc"
+)
+
+// Some host functions that are not implemented by Envoy end up getting imported anyways
+// by code that gets compiled but not executed at runtime. Because we know they are not
+// executed, we can stub them out to allow functioning on Envoy. Note, these match the
+// names and signatures of libc, not WASI ABI.
```

Comment:
Is this something we should report on tinygo?

Reply:
It might be but at the same time it's sort of unsustainable to guarantee imports don't end up in the binary in the long term I think. The problem is more on Envoy side for having an incomplete implementation of WASI, and even if that gets fixed in the future, supporting old versions here means this workaround will exist for a long time I think

Reply:
similarly step one in wasi support of wazero was to stub all the wasi functions to return ErrnoNosys(52). it removed the whack-a-mole problem of unused imports causing a module to never instantiate. Agree with rag that especially in envoy this only helps forward versions as wasm fixes are rarely back ported.